mongo/jvm/src/main/scala/com/avsystem/commons/mongo/typed/MongoUpdate.scala

```diff
@@ -45,9 +45,15 @@ sealed trait MongoUpdate[T] {
         case ArrayElementsQualifier.FirstMatching() => "$"
         case ArrayElementsQualifier.Each() => "$[]"
         case ArrayElementsQualifier.Filtered(filter) =>
-          val name = s"filter${arrayFilters.size}"
-          arrayFilters.add(Bson.document(name, filter.toBson))
-          s"$$[$name]"
+          val identifier = s"filter${arrayFilters.size}"
+          val (name, query) = filter match {
+            case MongoFilter.PropertyValueFilter(prop, filter) =>
```

Comment:
nit: `filter` is overloaded here with a different reference, not the most readable / error-prone